src/ddscxx/include/dds/sub/detail/Sample.hpp

```diff
@@ -111,9 +111,9 @@ class Sample
         return &this->data_;
     }
 
-    dds::sub::SampleInfo *info_ptr()
+    dds::sub::SampleInfo &info_ptr()
```

Comment:
Maybe rename this function to info_ref, since that is what it is doing now?

Reply:
You are right, it used to return a pointer but was changed to return a ref instead, but the name wasn't changed with it. However, I noticed that the info() function itself was already overloaded with a const and non-const getter, so there is not even a need for another function that returns the non-const pointer. I will remove info_ptr() and replace all its invocations with calls to the non-const getter of info() instead. I will do this both for Sample.hpp and for SampleRef.hpp.